skada/_reweight.py

```diff
@@ -669,3 +673,192 @@ def KLIEP(
+        The source data used for fitting.
+    `y_source_` : array-like, shape (n_samples,)
+        The source labels used for fitting.
+
+    References
+    ----------
+    .. [4] Kun Zhang et. al. Domain Adaptation under Target and Conditional Shift
+           In ICML, 2013.
+    """
+
+    def __init__(self, gamma):
+        super().__init__()
+        self.gamma = gamma
+
+    def fit(self, X, y, sample_domain=None, **kwargs):
+        """Fit adaptation parameters.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The source data.
+        y : array-like, shape (n_samples,)
+            The source labels.
+        sample_domain : array-like, shape (n_samples,)
+            The domain labels (same as sample_domain).
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, y, sample_domain = check_X_y_domain(X, y, sample_domain)
+        X_source, _, y_source, _ = source_target_split(
+            X, y, sample_domain=sample_domain
+        )
+        self.X_source_fit_, self.y_source_fit_ = X_source, y_source
+        return self
+
+    def adapt(self, X, y=None, sample_domain=None, **kwargs):
+        """Predict adaptation (weights, sample or labels).
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The source data.
+        y : array-like, shape (n_samples,)
+            The source labels.
+        sample_domain : array-like, shape (n_samples,)
+            The domain labels (same as sample_domain).
+
+        Returns
+        -------
+        X_t : array-like, shape (n_samples, n_components)
+            The data (same as X).
+        y_t : array-like, shape (n_samples,)
+            The labels (same as y).
+        weights : array-like, shape (n_samples,)
+            The weights of the samples.
+        """
+        X, sample_domain = check_X_domain(X, sample_domain)
+        X_source, X_target = source_target_split(X, sample_domain=sample_domain)
+
+        m = X_source.shape[0]
+        n = X_target.shape[0]
+
+        # if no source data, return uniform weights for target
+        if m == 0:
+            return AdaptationOutput(X=X, y=y, sample_weights=np.zeros(n))
+
+        # if no y provided, get the source labels from fit method
+        if y is None:
+            np.testing.assert_array_equal(
+                X_source,
+                self.X_source_fit_,
+                err_msg="Source data is different from the one used"
+                " for fitting and new labels are not provided."
+            )
+            y_source = self.y_source_fit_
+        else:
+            X, y = check_X_y_domain(X, y, sample_domain)
+            y_source, _ = source_target_split(y, sample_domain=sample_domain)
+
+        # check y is discrete or continuous
+        discrete = np.issubdtype(y_source.dtype, np.integer)
```

Comment:
maybe better to use either the` _find_y_type()` from `_utils.py `or `type_of_target()` from sklearn because some labels can be float but still discrete